dotCMS/src/main/java/com/dotcms/webdav/FsMemoryLockManager.java

```diff
@@ -0,0 +1,141 @@
+        locksByToken = new HashMap<String, CurrentLock>();
+    }
+
+    @Override
+    public synchronized LockResult lock( LockTimeout timeout, LockInfo lockInfo, LockableResource r ) {
+        FsResource resource = (FsResource) r;
+        LockToken currentLock = currentLock( resource );
+        if( currentLock != null ) {
+            return LockResult.failed( LockResult.FailureReason.ALREADY_LOCKED );
+        }
+
+        LockToken newToken = new LockToken( UUID.randomUUID().toString(), lockInfo, timeout );
+        CurrentLock newLock = new CurrentLock( resource.getFile(), newToken, lockInfo.lockedByUser );
+        locksByFile.put( resource.getFile(), newLock );
+        locksByToken.put( newToken.tokenId, newLock );
+        return LockResult.success( newToken );
+    }
+
+    @Override
+    public synchronized LockResult refresh( String tokenId, LockableResource resource ) {
+        CurrentLock curLock = locksByToken.get( tokenId );
+        if( curLock == null ) {
+          Logger.info(FileSystemResourceFactory.class, "can't refresh because no lock");
+            return LockResult.failed( LockResult.FailureReason.PRECONDITION_FAILED );
+        } else {
+            curLock.token.setFrom( new Date() );
+            return LockResult.success( curLock.token );
+        }
+    }
+
+    @Override
+    public synchronized void unlock( String tokenId, LockableResource r ) throws NotAuthorizedException {
+        FsResource resource = (FsResource) r;
+        LockToken lockToken = currentLock( resource );
+        if( lockToken == null ) {
+          Logger.info(FileSystemResourceFactory.class, "not locked" );
+            return;
+        }
+        if( lockToken.tokenId.equals( tokenId ) ) {
+            removeLock( lockToken );
+        } else {
+            throw new NotAuthorizedException( resource );
+        }
+    }
+
+    private LockToken currentLock( FsResource resource ) {
+        CurrentLock curLock = locksByFile.get( resource.getFile() );
+        if( curLock == null ) return null;
+        LockToken token = curLock.token;
+        if( token.isExpired() ) {
+            removeLock( token );
+            return null;
+        } else {
+            return token;
+        }
+    }
+
+    private void removeLock( LockToken token ) {
+      Logger.info(FileSystemResourceFactory.class, "removeLock: " + token.tokenId );
+        CurrentLock currentLock = locksByToken.get( token.tokenId );
+        if( currentLock != null ) {
+            locksByFile.remove( currentLock.file );
+            locksByToken.remove( currentLock.token.tokenId );
+        } else {
+          Logger.info(FileSystemResourceFactory.class,"couldnt find lock: " + token.tokenId );
+        }
+    }
+
+    @Override
+    public LockToken getCurrentToken( LockableResource r ) {
+        FsResource resource = (FsResource) r;
+        CurrentLock lock = locksByFile.get( resource.getFile() );
+        if( lock == null ) return null;

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00121)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00122)